src/hooks/web3/useJsonRpcUrlMap.tsx

```diff
@@ -30,3 +15,38 @@ export default function useJsonRpcUrlMap(): Record<SupportedChainId, string[]> {
   invariant(jsonRpcUrlMap, 'useJsonRpcUrlMap used without initializing the context')
   return jsonRpcUrlMap
 }
+
+function toJsonRpcMap(connectionMap?: JsonRpcConnectionMap): Record<SupportedChainId, (JsonRpcProvider | string)[]> {
+  return ALL_SUPPORTED_CHAIN_IDS.reduce((map, chainId) => {
+    const value = connectionMap?.[chainId]
+    const connections = (Array.isArray(value) ? value : [value])
+      .filter((value): value is string | JsonRpcProvider => Boolean(value))
+      .concat(...JSON_RPC_FALLBACK_ENDPOINTS[chainId])
+    return {
+      ...map,
+      [chainId]: connections,
+    }
+  }, {} as Record<SupportedChainId, (JsonRpcProvider | string)[]>)
```

Comment:
is there an alternative to casting here? someone could change the logic and break this in the future